lab_6_pipeline/pipeline.py

```diff
@@ -18,16 +31,33 @@ def __init__(self, path_to_raw_txt_data: Path):
         """
         Initializes CorpusManager
         """
+        self._scan_dataset()
+        self._validate_dataset()
+        self._storage = {}
+        self.path_to_raw_txt_data = Path(path_to_raw_txt_data)
 
     def _validate_dataset(self) -> None:
         """
         Validates folder with assets
         """
+        if not self.path_to_raw_txt_data.exists():
+            raise FileNotFoundError
+
+        if not self.path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError
+
+        if self.path_to_raw_txt_data.stat().st_size == 0:
+            raise EmptyDirectoryError
+
+        if len(list(self.path_to_raw_txt_data.glob("*.json"))) != len(list(self.path_to_raw_txt_data.glob("*.txt"))):
+            raise InconsistentDatasetError
 
     def _scan_dataset(self) -> None:
         """
         Register each dataset entry
         """
+        for file in self.path_to_raw_txt_data.glob('*'):
```

Comment:
does '*' option allows you to get all articles correctly?

Reply:
changed to glob('*.txt'). List with names of files sorted by descending order.

Reply:
UPD: changed completely, using ```get_article_id_from_filepath`` and ```from_raw``` functions